frontend/lib/shared/payment_details_page.dart

```diff
@@ -0,0 +1,179 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_starter/l10n/app_localizations.dart';
+import 'package:flutter_starter/shared/grid.dart';
+import 'package:flutter_starter/shared/json_schema_form.dart';
+import 'package:flutter_starter/shared/payment_method.dart';
+import 'package:flutter_starter/shared/search_payment_methods_page.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+class PaymentDetailsPage extends HookConsumerWidget {
+  const PaymentDetailsPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final paymentMethods = ref.watch(paymentMethodProvider);
+
+    final paymentTypes =
+        paymentMethods?.map((method) => method.kind.split('_').first).toSet();
```

Comment:
There's a `first` in here, you want that?

Reply:
https://github.com/TBD54566975/didpay/pull/47#discussion_r1482277651

^ makes no difference between `first` and `firstOrNull` but doesn't hurt to be consistent, will switch it up